src/app/services/error/error-message.service.ts

```diff
@@ -43,14 +47,47 @@ export class ErrorMessageService {
 
   //Used to handle other components/endpoints that do not subscribe to the error
   private handleOtherPages(error: HttpErrorResponse) {
-    const isIncluded = subscribedEndpoints.some((endpoint) => error.url?.includes(endpoint));
-    if (error.status === 500 && !isIncluded) {
-      this.showInternalError();
+    if (!this.isSubscribed(error)) {
+      switch (error.status) {
+        case 403:
+          this.showForbidden();
+          break;
+        case 404:
+          this.showNotFound();
+          break;
+        case 500:
+          this.showInternalError();
+          break;
+        default:
+          this.showInternalError();
+          break;
+      }
     }
   }
 
-  private showInternalError() {
-    this.router.navigateByUrl('internal-error');
+  private isSubscribed(response: HttpErrorResponse) {
+    return subscribedEndpoints.some((subscribed) => {
+      const endpointMatches = response.url?.includes(subscribed.endpoint);
+      const responseMatches = subscribed.responses.includes(response.status);
+
+      return endpointMatches && responseMatches;
+    });
+  }
+
+  private showGlobalErrorPage(route: string) {
+    this.router.navigateByUrl(route);
     setTimeout(() => this.headerService.hideNavigation(), 0);
```

Comment:
Is there a specific reason for the `setTimeout`?

Reply:
Annoyingly it wouldn't work without setTimeout due to showNavigation being applied on URL changes by default (line 30 app.component.ts). So using setTimeout pushed it out to the next event loop to hideNavigation after.